patroni/config.py

```diff
@@ -400,6 +400,58 @@ def save_cache(self) -> None:
                     except Exception:
                         logger.error('Can not remove temporary file %s', tmpfile)
 
+    def _validate_and_adjust_timeouts(self, config: Dict[str, Any]) -> None:
+        """Validate and adjust ``loop_wait``, ``retry_timeout``, and ``ttl`` values if necessary.
+
+        Minimal values:
+
+            * ``ttl``: 20 seconds;
+            * ``loop_wait``: 1 second;
+            * ``retry_timeout``: 3 seconds.
+
+        Maximum values:
+        In case if values don't fulfill the following rule, ``retry_timeout`` and ``loop_wait``
+        are reduced so that the rule is fulfilled:
+
+            .. code-block:: python
+
+                loop_wait + 2 * retry_timeout <= ttl
+
+        .. note:
+            We prefer to reduce ``loop_wait`` and will reduce ``retry_timeout`` only if ``loop_wait``
+            is already set to a minimal possible value.
+
+        :param config: :class:`dict` object with global configuration.
+        """
+        min_ttl = 20
+        ttl = config.get('ttl', self.__DEFAULT_CONFIG['ttl'])
+        if ttl < min_ttl:
```

Comment:
so we are sure `ttl` is always int here?

Reply:
Nice catch. Actually the whole method should be called from the try block. https://github.com/zalando/patroni/pull/2869/commits/b71ebc1ac73c6961894c494e1266989515819aa5 addressing it.